solutions/tests/test_is_leap_year.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Test module for is_leap function.
+Contains intentionally buggy tests for debugging practice.
+
+Test categories:
+    - Standard cases: Common leap and non-leap years
+    - Edge cases: Smallest leap year, non-leap years close to leap years
+    - Defensive tests: Handling of invalid inputs (e.g., non-integer, negative values)
+
+Created on XX XX XX
```

Comment:
add date

Reply:
@YonatanBest also mentioned this, I'll update it on my other challenge as well 👍